language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SortingUtil.java

```diff
@@ -215,11 +218,61 @@ public static String genSortTextForTypeDescContext(BallerinaCompletionContext co
         if (item.getType() == SNIPPET && ((SnippetCompletionItem) item).kind() == SnippetBlock.Kind.TYPE) {
             return genSortText(8);
         }
-        
+
         // Case 7+ 
         return genSortText(SortingUtil.toRank(item, 8));
     }
 
+    /**
+     * Generate the sort text based on the assignability to particular type symbol.
+     * <p>
+     * Sorting order is defined as follows
+     * 1.Variables assignable to the given type symbol
+     * 2.Functions with return type descriptor assignable to given type symbol
+     * 3.Other symbols follows the default sorting order
+     *
+     * @param completionItem Completion item.
+     * @return {@link String} Sort text.
+     */
+    public static String genSortTextByAssignability(LSCompletionItem completionItem, TypeSymbol typeSymbol) {
+        int rank = SortingUtil.toRank(completionItem, 2);
+        if (completionItem.getType() == LSCompletionItem.CompletionItemType.SYMBOL) {
+            Optional<Symbol> completionSymbol = ((SymbolCompletionItem) completionItem).getSymbol();
+            if (completionSymbol.isEmpty()) {
+                return SortingUtil.genSortText(rank);
+            }
+            Optional<TypeSymbol> evalTypeSymbol = SymbolUtil.getTypeDescriptor(completionSymbol.get());
+            if (evalTypeSymbol.isPresent()) {
+                switch (completionItem.getCompletionItem().getKind()) {
```

Comment:
Why do we check the completion item kind here? Any specific reason for that?

Reply:
We have to give highest precedence to vars and then to functions. Followed the default implementation here https://github.com/ballerina-platform/ballerina-lang/blob/1d1c1e80234a2dce16f789d0e0ea7e16aeff2325/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SortingUtil.java#L340-L346

Reply:
Yeah, we don't need to duplicate the logic here. Since the sorting text generation can be used in a nested manner you can first assign your priority with your sort text (Say 1 -> A). This creates a group. Now, you can re-order within the group as well. Then it will contain a sort text with two syllables, like (AA, AB, AC, etc). In order to do that, you can combine two sort texts as done in https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ListenerDeclarationNodeContext.java#L114

Reply:
Done!